### DISCUSSION TITLE
Problems with list HTML

### DISCUSSION THREAD

**Discussion by madviking** [Laravel-Backpack/community-forum#480] [Q&A (Help)]
I'm struggling a lot with a customised list view, due to list generation being all over the files & not organised or documented well. I have two problems with this view that seem very trivial, but are anything but.  

1) "empty" row before the actual data
2) hidden popup

*For the first one (problem area marked with red):*
**
![image](https://user-images.githubusercontent.com/167365/236619088-14095d14-07bf-4b85-9803-22e8c4bc6376.png)
**

Now if I try to hide it with this:
`.dataTables_scrollBody thead tr:nth-child(1) {
            display: none;
        }`
        

it does hide it, but it will also mess up the columns:
**
<img width="1232" alt="image" src="https://user-images.githubusercontent.com/167365/236619126-c1f38d57-7b00-403d-bd51-ef3a50c6c6a4.png">
**

*2) The popup - popup is inside the container so it is hidden if the table only has one row / it's the last row. Setting z-index does not help.*
<img width="563" alt="image" src="https://user-images.githubusercontent.com/167365/236619194-ce237664-e2d1-4264-acb9-2b06dea226b9.png">

<img width="563" alt="image" src="https://user-images.githubusercontent.com/167365/236619199-232ef0d4-a5ad-4fd0-9ac2-9d0b2615686a.png">

Now if I edit datatables_logic.blade.php and remove the dom code from line 251 entirely, it shows the popup ok, but ofc everything else breaks. 

<img width="507" alt="image" src="https://user-images.githubusercontent.com/167365/236619275-f1b8a1dd-fd05-451a-968f-669b49f69d64.png">

This is the beautiful and rather indecipherable code part from the dom:
dom:
`            "<'row hidden'<'col-sm-6'i><'col-sm-6 d-print-none'f>>" +
              "<'row'<'col-sm-12'tr>>" +
              "<'row mt-2 d-print-none '<'col-sm-12 col-md-4'l><'col-sm-0 col-md-4 text-center'B><'col-sm-12 col-md-4 'p>>",
`

@tabacitu any useful pointers? 

I'm really loving backpack, but the choice of keeping it without proper Javascript FE framework really shows its ugly face on issues like this.

**Comment by tabacitu** [comment]
Yeah customizing DataTables can be a pain! It's the most feature-packed library for this, but... it's not pretty at all. I'm afraid there isn't much I can help with here. But you could take a look at Backpack v6's CSS file for customizations - maybe you find something there that's helpful to you - https://github.com/Laravel-Backpack/CRUD/blob/v6/src/resources/assets/css/common.css - the "extra row" problem was one we also encountered so you might find a fix there.

That being said, in Backpack v6 we're adding full theme support:
- instead of customizing _everything_ like now, you'll start from a "base theme" and only customize what you need;
- you'll have a clear process of how to create a custom theme;
- creating a new theme should take 4-5 hours MAX;

Until then though, I'm afraid there's not much I can help with, it's custom code.

Cheers!
